src/MySqlConnector/Core/SchemaProvider.cs

```diff
@@ -33,6 +34,21 @@ public DataTable GetSchema(string collectionName)
 			return dataTable;
 		}
 
+		private void FillRestrictions(DataTable dataTable)
+		{
+				dataTable.Columns.AddRange(new[] {
+					new DataColumn("RestrictionName", typeof(string)),
+					new DataColumn("ParameterName", typeof(string)),
+					new DataColumn("RestrictionDefault", typeof(string)),
+					new DataColumn("RestrictionNumber", typeof(int))
+				});
+
+				dataTable.Rows.Add("Catalog", "@Catalog", "TABLE_CATALOG", 1);
+				dataTable.Rows.Add("Owner", "@Owner", "TABLE_SCHEMA", 2);
+				dataTable.Rows.Add("Table", "@Name", "TABLE_NAME", 3);
+				dataTable.Rows.Add("TableType", "@TableType", "TABLE_TYPE", 4);
+		}
```

Comment:
The `Restrictions` schema should have four columns: `CollectionName`, `RestrictionName`, `RestrictionDefault`, `RestrictionNumber`: https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/common-schema-collections#restrictions

This table is filled with the names of restrictions that are supported by _other_ schema tables.

For example, to indicate three restrictions on the `Tables` schema, you would add these rows:



CollectionName | RestrictionName | RestrictionDefault | RestrictionNumber
-- | -- | -- | --
Tables | Schema |   | 1
Tables | Table |   | 2
Tables | TableType |   | 3

While it's not OK to bring code from Connector/NET into this project, it is totally fine to perform "black box" testing and see what (for example) `connection.GetSchema("Restrictions")` returns when executed against that library. This code should return a subset of what it supports.